lib/MediaConnection/RTCPeerMediaConnection.js

```diff
@@ -0,0 +1,171 @@
+const EventEmitter = require('events').EventEmitter;
+
+module.exports = class RTCPeerMediaConnection extends EventEmitter {
+
+    constructor(pcConfig, iceConnectionStateCallback)
+    {
+        super();
+        this.rtcPeerConnection = new RTCPeerConnection(pcConfig);
+        this.rtcPeerConnection.addEventListener('iceconnectionstatechange', () => {
+            iceConnectionStateCallback(this.rtcPeerConnection.iceConnectionState)
+        });
+    }
+
+
+    createSdpOffer(options) {
+
+        return this.rtcPeerConnection.signalingState === 'stable' ? this.rtcPeerConnection.createOffer(options)
+            .then(offer => {
+                return this.setLocalSdp(offer);
+            }) : Promise.resolve();
+    }
+
+    createSdpAnswer(options) {
+
+        return this.rtcPeerConnection.createAnswer(options).then(offer => {
+            this.setLocalSdp(offer)
+            return offer;
+        });
+    }
+
+    setRemoteSdp(type, sdp)
```

Comment:
Let's use `RTCSessionDescription` as argument please. It's quite a universal object type.

Reply:
The RTCPeerConnection.createOffer use the 'RTCOfferOptions'-Type for the parameter? Should i use 'rtcOfferOptions' as parameter?

Reply:
> Let's use RTCSessionDescription as argument please. It's quite a universal object type.

I was referring to `setRemoteSdp()`

> The RTCPeerConnection.createOffer use the 'RTCOfferOptions'-Type for the parameter? Should i use 'rtcOfferOptions' as parameter?

I guess you are referring to `createSdpOffer()`. If so, yes an optional `options: RTCOfferOptions`  argument would make it.